scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
Hmm, ok, I'm not sure whether it makes more sense to output an empty array or raise an exception in this case.

The code in my previous comment checks whether dimensions are violated in both directions, and just has one extra check compared to the `_is_valid_mode_shapes()` that your code is currently using.